Client/Extensions/UIAlertControllerExtensions.swift

```diff
@@ -87,6 +87,30 @@ extension UIAlertController {
         return alert
     }
 
+    class func clearPrivateDataAlert(okayCallback: () -> Void) -> UIAlertController {
+        let alert = UIAlertController(
+            title: NSLocalizedString("Clear Private Data", tableName: "ClearPrivateDataConfirm", comment: "Title of the confirmation dialog shown when a user tries to clear private data."),
+            message: NSLocalizedString("This action will clear all of your private data. It cannot be undone.", tableName: "ClearPrivateDataConfirm", comment: "Description of the confirmation dialog shown when a user tries to clear their private data."),
+            preferredStyle: UIAlertControllerStyle.Alert
+        )
+
+        let noOption = UIAlertAction(
+            title: NSLocalizedString("Cancel", tableName: "ClearPrivateDataConfirm", comment: "The cancel button when confirming clear private data."),
+            style: UIAlertActionStyle.Cancel,
+            handler: nil
+        )
+
+        let okayOption = UIAlertAction(
+            title: NSLocalizedString("Clear", tableName: "ClearPrivateDataConfirm", comment: "The button that clears private data."),
+            style: UIAlertActionStyle.Destructive,
+            handler: {_ in okayCallback() }
```

Comment:
The perils of copy and paste programming. Sure :)

On 22 February 2016 at 14:59, Stephan Leroux <notifications@github.com>
wrote:

> In Client/Extensions/UIAlertControllerExtensions.swift
> <https://github.com/mozilla/firefox-ios/pull/1555#discussion_r53635476>:
>
> > +        let alert = UIAlertController(
> > +            title: NSLocalizedString("Clear Private Data", tableName: "ClearPrivateDataConfirm", comment: "Title of the confirmation dialog shown when a user tries to clear private data."),
> > +            message: NSLocalizedString("This action will clear all of your private data. It cannot be undone.", tableName: "ClearPrivateDataConfirm", comment: "Description of the confirmation dialog shown when a user tries to clear their private data."),
> > +            preferredStyle: UIAlertControllerStyle.Alert
> > +        )
> > +
> > +        let noOption = UIAlertAction(
> > +            title: NSLocalizedString("Cancel", tableName: "ClearPrivateDataConfirm", comment: "The cancel button when confirming clear private data."),
> > +            style: UIAlertActionStyle.Cancel,
> > +            handler: nil
> > +        )
> > +
> > +        let okayOption = UIAlertAction(
> > +            title: NSLocalizedString("Clear", tableName: "ClearPrivateDataConfirm", comment: "The button that clears private data."),
> > +            style: UIAlertActionStyle.Destructive,
> > +            handler: {_ in okayCallback() }
>
> Nit: Could you just pass in okayCallback instead of wrapping it in a
> closure?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/firefox-ios/pull/1555/files#r53635476>.
>